src/object/authenticate.c

```diff
@@ -5129,31 +5175,63 @@ au_change_owner (MOP classmop, MOP owner)
   else
     {
       error = au_fetch_class_force (classmop, &class_, AU_FETCH_UPDATE);
-      if (error == NO_ERROR)
+      if (error != NO_ERROR)
 	{
-	  /*
-	   * Change serial object's owner when the class has auto_increment
-	   * attribute column.
-	   */
-	  for (attr = class_->attributes; attr != NULL; attr = (SM_ATTRIBUTE *) attr->header.next)
+	  goto exit_on_error;
+	}
+
+      if (ws_is_same_object (class_->owner, owner))
+	{
+	  goto exit_on_error;
+	}
+
+      /*
+       * Change serial object's owner when the class has auto_increment
+       * attribute column.
+       */
+      for (attr = class_->attributes; attr != NULL; attr = (SM_ATTRIBUTE *) attr->header.next)
+	{
+	  if (attr->auto_increment != NULL)
 	    {
-	      if (attr->auto_increment != NULL)
+	      error = au_change_serial_owner (&attr->auto_increment, owner);
+	      if (error != NO_ERROR)
 		{
-		  error = au_change_serial_owner (&attr->auto_increment, owner);
-		  if (error != NO_ERROR)
-		    {
-		      goto exit_on_error;
-		    }
+		  goto exit_on_error;
 		}
 	    }
+	}
+
+      /* Change class owner */
+      class_->owner = owner;
+
+      /*
+       * Since class_full_name includes owner_name, if owner of class is changed,
+       * class_full_name must be changed as well.
+       */
+      class_name = sm_ch_simple_name ((MOBJ) class_);
 
-	  /* Change class owner */
-	  class_->owner = owner;
+      if (sm_check_system_class_by_name (class_name))
```

Comment:
If (sm_check_system_class_by_name (class_name)) is not satisfied, do I not have to perform locator_flush_class (classmop)?

Reply:
I understand your explanation.
thank you.